src/rules/declaration-empty-line-before/index.js

```diff
@@ -0,0 +1,101 @@
+import {
+  blockString,
+  isCustomProperty,
+  isSingleLineString,
+  isStandardSyntaxDeclaration,
+  isStandardSyntaxProperty,
+  optionsHaveException,
+  optionsHaveIgnored,
+  report,
+  ruleMessages,
+  validateOptions,
+} from "../../utils"
+
+export const ruleName = "declaration-empty-line-before"
+
+export const messages = ruleMessages(ruleName, {
+  expected: "Expected empty line before declaration",
+  rejected: "Unexpected empty line before declaration",
+})
+
+export default function (expectation, options) {
+  return (root, result) => {
+    const validOptions = validateOptions(result, ruleName, {
+      actual: expectation,
+      possible: [
+        "always",
+        "never",
+      ],
+    }, {
+      actual: options,
+      possible: {
+        except: [
+          "first-nested",
+          "after-comment",
+          "after-declaration",
+        ],
+        ignore: [
+          "inside-single-line-block",
+        ],
+      },
+      optional: true,
+    })
+    if (!validOptions) { return }
+
+    root.walkDecls(decl => {
+      const { prop, parent } = decl
+
+      if (!isStandardSyntaxDeclaration(decl)) { return }
+      if (!isStandardSyntaxProperty(prop)) { return }
```

Comment:
Ok, I'm convinced.

We can drop the use of `isStandardSyntaxProperty` here. And then add

```js
// SCSS var (e.g. $var: x), list (e.g. $list: (x)) or map (e.g. $map: (key:value))
if (decl.prop[0] === "$") { return false }

// Less var (e.g. @var: x)
if (decl.prop[0] === "@") { return false }
```

To `isStandardSyntaxDeclaration`. That will ignore SCSS-like variables, maps etc and Less variables (but not declaration that use variable *interpolation* in their property name).